runtime/src/iree/hal/drivers/hip/hip_device.c

```diff
@@ -89,6 +89,151 @@ typedef struct iree_hal_hip_device_t {
+  return IREE_HIP_RESULT_TO_STATUS(table->hip_symbols, hipEventCreate(out),
+                                   "hipEventCreate");
+}
+iree_status_t iree_hal_hip_deferred_work_queue_symbol_table_record_native_event(
+    iree_hal_deferred_work_queue_device_interface_t* symbol_table,
+    void* event) {
+  iree_hal_hip_deferred_work_queue_device_interface_t* table =
+      (iree_hal_hip_deferred_work_queue_device_interface_t*)(symbol_table);
+  hipEvent_t hip_event = (hipEvent_t)(event);
+  return IREE_HIP_RESULT_TO_STATUS(
+      table->hip_symbols, hipEventRecord(hip_event, table->dispatch_hip_stream),
+      "hipEventRecord");
+}
+
+iree_status_t
+iree_hal_hip_deferred_work_queue_symbol_table_synchronize_native_event(
+    iree_hal_deferred_work_queue_device_interface_t* symbol_table,
+    void* event) {
+  iree_hal_hip_deferred_work_queue_device_interface_t* table =
+      (iree_hal_hip_deferred_work_queue_device_interface_t*)(symbol_table);
+  hipEvent_t hip_event = (hipEvent_t)(event);
+  return IREE_HIP_RESULT_TO_STATUS(table->hip_symbols,
+                                   hipEventSynchronize(hip_event));
+}
+iree_status_t
+iree_hal_hip_deferred_work_queue_symbol_table_destroy_native_event(
+    iree_hal_deferred_work_queue_device_interface_t* symbol_table,
+    void* event) {
+  iree_hal_hip_deferred_work_queue_device_interface_t* table =
+      (iree_hal_hip_deferred_work_queue_device_interface_t*)(symbol_table);
+  hipEvent_t hip_event = (hipEvent_t)(event);
+  return IREE_HIP_RESULT_TO_STATUS(table->hip_symbols,
+                                   hipEventDestroy(hip_event));
+}
+
+iree_status_t
+iree_hal_hip_deferred_work_queue_symbol_table_semaphore_acquire_timepoint_device_signal_native_event(
+    iree_hal_deferred_work_queue_device_interface_t* symbol_table,
+    struct iree_hal_semaphore_t* semaphore, uint64_t value, void** out_event) {
+  hipEvent_t* out = (hipEvent_t*)(out_event);
+  return iree_hal_hip_event_semaphore_acquire_timepoint_device_signal(
+      semaphore, value, out);
+}
+
+bool iree_hal_hip_deferred_work_queue_symbol_table_acquire_host_wait_event(
+    iree_hal_deferred_work_queue_device_interface_t* symbol_table,
+    struct iree_hal_semaphore_t* semaphore, uint64_t value, void** out_event) {
+  return iree_hal_hip_semaphore_acquire_event_host_wait(
+      semaphore, value, (iree_hal_hip_event_t**)out_event);
+}
+
+void iree_hal_hip_deferred_work_queue_symbol_table_release_wait_event(
+    iree_hal_deferred_work_queue_device_interface_t* symbol_table,
+    void* wait_event) {
+  iree_hal_hip_event_release(wait_event);
+}
+
+void* iree_hal_hip_deferred_work_queue_symbol_table_native_event_from_wait_event(
+    iree_hal_deferred_work_queue_device_interface_t* symbol_table,
+    void* event) {
+  iree_hal_hip_event_t* wait_event = (iree_hal_hip_event_t*)event;
+  return iree_hal_hip_event_handle(wait_event);
+}
+
+iree_status_t
+iree_hal_hip_deferred_work_queue_symbol_table_create_command_buffer_for_deferred(
```

Comment:
maybe `create_stream_command_buffer` or `create_immediate_command_buffer` (it's used by deferred playback, but it's really just an immediately executing command stream)